src/QsCompiler/Tests.Compiler/TestCases/TypeParameterResolution.qs

```diff
@@ -157,3 +157,110 @@ namespace Microsoft.Quantum.Testing.TypeParameterResolution {
 
 // Foo.A -> Int
 // Bar.B -> Int
+
+// =================================
+
+// Basic Entry Point
```

Comment:
Does this test adds value? The "unrelated" test below will check that @EntryPoint does trigger call tree creation and the tests without @EntryPoint already validate various call structures. No?

Reply:
This tests that the call graph is generated as expected with an entry point. It tests multiple directed descendants as well as indirect descendants. You are correct in pointing out that these two things are already checked by tests that do not use entry points, but this test makes sure that those behaviors work with entry points as well.

Reply:
Separated From Entry Point by Specialization tests a special case for entry points. Only when entry points are in the graph do we care about the question of if two nodes are related to each other. This test answers the question of does that relation persist across different specializations of the same callable.